### PR TITLE
Fix for file names with spaces; Form re-sizing, show progress, added file menu

### PR DESCRIPTION
**Pull request by BAD-AL**
Form re-sizing
 no progress shown
 filenames with spaces don't work
Added:
 File Menu
 'Clear Items' functionality

### CODE REVIEW

**Comment by BAD-AL** [comment]
I do actually feel the gui would be better as a .net framework app because:
1. they can compile to a single .exe
2. Everyone who has windows has the .net framework; .net core/new .net has a new version come out every 5 minutes and not everyone has all the versions.
But if you like using the .net core, then np.

File: gui/Form1.cs [hunk 1]
```diff
@@ -63,12 +63,29 @@ private void pickFolder_Click(object sender, EventArgs e)
...
+			else
+			{
+				repackProgress.Value++;
+			}
+		}
+
 		private async void repackButton_Click(object sender, EventArgs e)
 		{
+			List<string> files = new List<string>();
+			foreach (var v in lvlsList.Items) files.Add(v.ToString());
```
Comment (PrismaticFlower) [inline gui/Form1.cs]: The compiler seems to be issuing a warning "Possible null reference argument for parameter 'item' in 'void List<string>.Add(string item)'." for this line. I'm guessing it just can't see that it's never null or is there actually the possibility it'll be null?

**Comment by PrismaticFlower** [comment]
Nice! I've left one question in the comments about a compiler warning. Aside from that to me it all looks good.

> I do actually feel the gui would be better as a .net framework app because:
> 
>     1. they can compile to a single .exe
> 
>     2. Everyone who has windows has the .net framework; .net core/new .net has a new version come out every 5 minutes and not everyone has all the versions.
>        But if you like using the .net core, then np.


Nah, I don't really care what it uses. I think it's just on whatever Visual Studio defaulted to, I have no issue with changing it's target.

**Comment by BAD-AL** [comment]
new size for lvl-repack.exe => 48 KB
now just need to distribute 2 .exes side-by-side; none of the extra .net core files needed.

File: gui/Form1.cs [hunk 2]
```diff
@@ -1,113 +1,194 @@
...
+			if (pickDialog.ShowDialog() == DialogResult.OK)
+			{
+				folderPath = pickDialog.SelectedPath;
+			}
+			pickDialog.Dispose();
 
-			pickDialog.SetOptions(FILEOPENDIALOGOPTIONS.FOS_PATHMUSTEXIST | FILEOPENDIALOGOPTIONS.FOS_FILEMUSTEXIST | FILEOPENDIALOGOPTIONS.FOS_NOCHANGEDIR | FILEOPENDIALOGOPTIONS.FOS_PICKFOLDERS);
-			pickDialog.SetTitle("Pick Folder");
-			pickDialog.SetOkButtonLabel("Pick");
+			AddPath( folderPath);
```
Comment (PrismaticFlower) [inline gui/Form1.cs]: Ah, I didn't realize we'd have to change this in order to use .NET Framework. The legacy folder picker is too huge a step back in usability. (For me at least.) I'd rather have the downsides of .NET 8 than it.

**Comment by PrismaticFlower** [comment]
Unfortunately this seems to have broken the project files for me. And there are now two solutions? And wow that a lot of changes haha. It's getting hard for me to follow them all. How difficult would it be to separate some of them out?

File: gui/Form1.cs (hunk 2 above)
Comment (BAD-AL) [reply gui/Form1.cs]: So, the usability improvement would be the Drag and drop. Could use 'pick folder' dialog if you really wanted.

**Comment by BAD-AL** [comment]
ok, I'll remove some un-necessary files and organize the form a little more.

Also, the gui is built with .NET Framework 4.6 (oldest .net framework sdk you can still easily get)
   https://dotnet.microsoft.com/en-us/download/visual-studio-sdks

You may need to install it. 
It is kindof a pain to have to install from the development side, but for users it's something they will be able to run without downloading anything more.

File: gui/Form1.cs (hunk 2 above)
Comment (PrismaticFlower) [reply gui/Form1.cs]: Drag and Drop is a very nice addition and it provides another good workflow. For me it doesn't change the need for IFileOpenDialog to be available as well. In theory we should be able to have it both ways, use IFileOpenDialog and .NET Framework but I don't know C# well enough to know where to start with manual COM interop.

Which isn't very helpful, if I had time I would learn how and do it myself. It's probably out of scope for what you were wanting to do as well. Which I'm guessing is just to have a version people can download and have work out of the box. Towards that if it's simpler than sorting this out I don't mind at all if you'd prefer to maintain and distribute a fork without merging changes to the GUI back here.

Of course if you'd rather sort this out and get everything merged in one place I'm happy with that as well. Totally your call.

**Comment by BAD-AL** [comment]
I did a little more research and now a days you can publish stand-alone exes with the new .net.
The downside is that they end up being like 50-60 MB for a simple program (because they gotta pack in the runtime).

I can use the better file picker that you had in originally with .NET framework; but it would require the program to take along a bunch of other .dlls (which defeats the purpose of the switch to .NET Framework).

So I think the options are:
1. Keep the project as is on this PR; the user can use Drag and Drop to select files and folders they will be presented with the horrible .NET FolderBrowser when they choose 'pick folder'  but have a very small .exe and users won't have to download anything else to get the program to run.

2. Target .NET Framework, make the user use Drag and drop to select files and folders (remove FolderBrowser option) & have a very small .exe.

3. ~Compile for .net 8 standalone exe; and use the better Folder picker.~
  ~The user can still user Drag and drop but the .exe will be like 60 MB.~
  The standalone exe does get generated in this case, but does not work.

4. Compile for .net 8 runtime; and use the better Folder picker.
  The user can still user Drag and drop and they'll need the .net 8 runtime installed to use this program.

  
  What's your preference?

**Comment by PrismaticFlower** [comment]
Sorry for the late response. I was quite busy today. But yeah 4. is the option I prefer. In an ideal world WinForms would support NativeAoT compilation (which would be self contained and small) then we could just use that. But alas not to be yet.

File: gui/Program.cs
Comment (BAD-AL) [inline gui/Program.cs:1]: not sure why this couldn't compile; had to comment out.

**Comment by PrismaticFlower** [comment]
Looks good. I just pushed a couple tiny changes. I'll merge it now and make a new release later.

Also of course, thank you for making these changes and fixes!

**Pull request merged**